datafusion/src/physical_plan/mod.rs

```diff
@@ -51,6 +51,11 @@ pub trait RecordBatchStream: Stream<Item = ArrowResult<RecordBatch>> {
     /// Implementation of this trait should guarantee that all `RecordBatch`'s returned by this
     /// stream should have the same schema as returned from this method.
     fn schema(&self) -> SchemaRef;
+
+    /// Returns the current memory usage for this stream.
+    fn mem_used(&self) -> usize {
+        0
+    }
```

Comment:
This line adds a `mem_used` method in our essential `RecordBatchStream` trait.

A baby step to tracking Non-Limited-Operators' memory usage since I think `SendableRecordBatchStream` is the fundamental entity that holds memory during execution.  However, I didn't quite find a way to register these streams generated during `async execute` to our memory manager.

I would love to hear your thoughts. If considered not appropriate, I will remove it.

Reply:
After re-consider this for a while. I cannot think of a solution to register `RecordBatchStream` somewhere else since each stream is used through mutable reference. I don't think there's a way sharing it except for Arc<dyn RecordBatchStream ....>, which we have discussed earlier and is not acceptable.

I'm going to revert this last commit.